### DISCUSSION TITLE
Overwrite Primitive Types in Generated Output

### DISCUSSION THREAD

**Discussion by boennemann** [prisma/prisma#10135] [Q&A]
Hello 👋 

Let's imagine we're handling strings that have to follow a certain pattern.
TypeScript makes it easier to enforce something like this at compile time.

```ts
type PrefixedString = `prefix-${string}`
type PostfixedString = `${string}-postfix`
```

Coming from Sequelize it was simple to just set this on the model and thus get warnings/enforcement for this pattern at every interaction with the model.

```ts
class ExampleModel extends Model {
  //...
  @Column(Types.STRING)
  exampleColumn: PrefixedString | PostfixedString
  //...
}
```

With Prisma I'm missing the possibility to overwrite certain primitive types in the generated output.

Unfortunately is wasn't very easy to find the right search terms here. I've only found details on how to overwrite the native db data type.

Is this something that's supported? Could this be considered?
Is there a workaround you could think of?

Thank you 🙏

**Comment by austincrim** [comment]
Hey @boennemann 👋, thanks for the question.

We currently don't support a pattern like this but if you would like to see it added to Prisma, please create a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) so our engineering teams can take a look 👍.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
